BraveShared/Analytics/DAU.swift

```diff
@@ -38,6 +40,11 @@ public class DAU {
             return false
         }
         
+        if processingPing {
```

Comment:
This is redundant once you make the DAU class object global. The previous check also do the same thing. Basically you cant have a sync check for a async updation. Also the check below for launchTimer != nil will make sure no new timers are created.